Dapper.Contrib/SqlMapperExtensions.cs

```diff
@@ -418,26 +418,35 @@ private static string GetTableName(Type type)
         /// Specifies a custom callback that detects the database type instead of relying on the default strategy (the name of the connection type object).
         /// Please note that this callback is global and will be used by all the calls that require a database specific adapter.
         /// </summary>
-        public static GetDatabaseTypeDelegate GetDatabaseType;
-
-
-        private static ISqlAdapter GetFormatter(IDbConnection connection)
+        [Obsolete("Set Dapper.Config.GetDbmsTypeFromConnection instead, this gets used for more than just extensions")]
+        public static GetDatabaseTypeDelegate GetDatabaseType

```

Comment:
No problems putting the getter back.

In terms of the obsolete flag, I can remove it until a later version if requited but I'm not too sure why this is an issue. The obsolete flag is only a compile time hint there is a newer way of doing this rather than a breaking change.